lib/commands/run.coffee

```diff
@@ -135,22 +135,27 @@ maybeInspectContainer = (docker, name) ->
 # Helper method to check and see if the running container's links differ from the ones we would
 # want to create it with. If they do differ, the only recourse is to remove and recreate the
 # container, since you may not (currently) modify links of a running container.
-isLinkMissing = (containerInfo, createOpts) ->
-  # These are the links as reported back by Docker, which have the format:
-  #  /sourceContainer:/destContainer/alias
-  # We do a bit of splitting and joining to convert them back to /sourceContainer:alias so that
-  # we can compare them directly with the Links parameter we provide to create.
-  currentLinks = _.map (containerInfo?.HostConfig?.Links or []), (link) ->
-    [source, dest] = link.split(':')
-    "#{source}:#{dest.split('/').pop()}"
+isLinkMissing = (containerInfo, createOpts, servicesConfig) ->
+  service = containerInfo.Name.split("/")[1].split(".")[0]
+  env = containerInfo.Name.split("/")[1].split(".")[1]
```

Comment:
You're splitting on . and only taking element 1.

Also, thinking about it, a few things:

You still need to check the current links because of Docker 1.9 and below
support.

You should check the create date of whatever containers are currently being
linked to, if the above check succeeds. No need to pull back in the prereq
calculations, assume that the container was created with the links it
needs.

On Thursday, June 23, 2016, Sam Neubardt <notifications@github.com> wrote:

> In lib/commands/run.coffee
> <https://github.com/twitter-fabric/galley/pull/44#discussion_r68281584>:
>
> > @@ -135,22 +135,27 @@ maybeInspectContainer = (docker, name) ->
> >  # Helper method to check and see if the running container's links differ from the ones we would
> >  # want to create it with. If they do differ, the only recourse is to remove and recreate the
> >  # container, since you may not (currently) modify links of a running container.
> > -isLinkMissing = (containerInfo, createOpts) ->
> > -  # These are the links as reported back by Docker, which have the format:
> > -  #  /sourceContainer:/destContainer/alias
> > -  # We do a bit of splitting and joining to convert them back to /sourceContainer:alias so that
> > -  # we can compare them directly with the Links parameter we provide to create.
> > -  currentLinks = _.map (containerInfo?.HostConfig?.Links or []), (link) ->
> > -    [source, dest] = link.split(':')
> > -    "#{source}:#{dest.split('/').pop()}"
> > +isLinkMissing = (containerInfo, createOpts, servicesConfig) ->
> > +  service = containerInfo.Name.split("/")[1].split(".")[0]
> > +  env = containerInfo.Name.split("/")[1].split(".")[1]
>
> Won't this be fine? If containerInfo.Name is www.test.cucumber then
> service would be www and env would be test.cucumber
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/twitter-fabric/galley/pull/44/files/c4529c41915573bf66832c9d42f2c95bcd74fdb2#r68281584>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAP65U44RuUrDQMAT8lU156u8hsSwdmSks5qOsbogaJpZM4I8Wb->
> .
>